compiler/z/codegen/OMRCodeGenerator.cpp

```diff
@@ -4754,38 +4754,35 @@ bool OMR::Z::CodeGenerator::getSupportsOpCodeForAutoSIMD(TR::ILOpCode opcode, TR
 
    TR::DataType et = opcode.getVectorResultDataType().getVectorElementType();
 
+   TR_ASSERT_FATAL(et == TR::Int8 || et == TR::Int16 || et == TR::Int32 || et == TR::Int64 || et == TR::Float || et == TR::Double, "DataType %s is not supported for vectorization", et.toString());
+
    // implemented vector opcodes
    switch (opcode.getVectorOperation())
       {
       case OMR::vadd:
       case OMR::vsub:
-      case OMR::vmul:
-      case OMR::vdiv:
-      case OMR::vneg:
-         if (et == TR::Int32 || et == TR::Int64 || et == TR::Float || et == TR::Double)
-            return true;
-         else
-            return false;
       case OMR::vload:
       case OMR::vloadi:
       case OMR::vstore:
       case OMR::vstorei:
-         if (et == TR::Int32 || et == TR::Int64 || et == TR::Float || et == TR::Double)
+      case OMR::vneg:
+      case OMR::vsplats:
+         return true;
+      case OMR::vmul:
+         if (et == TR::Int8 || et == TR::Int16 || et == TR::Int32 || et == TR::Float || et == TR::Double)
             return true;
          else
             return false;
-      case OMR::vxor:
-      case OMR::vor:
-      case OMR::vand:
-         if (et == TR::Int32 || et == TR::Int64)
+      case OMR::vdiv:
+         if (et == TR::Float || et == TR::Double)
             return true;
          else
             return false;
-      case OMR::vsplats:
-         if (et == TR::Int32 || et == TR::Int64 || et == TR::Float || et == TR::Double)
+      case OMR::vxor:
+      case OMR::vor:
+      case OMR::vand:
+         if (et == TR::Int8 || et == TR::Int16 || et == TR::Int32 || et == TR::Int64)
             return true;
-         else
-            return false;
```

Comment:
Previously for `vxor`, `vor`, and `vand`, there was an else clause to return false if `et` didn't match the supported types. Is that not needed anymore? Or is it because as the code stands, it will fall through to the `default` case; if so, then either you should add a comment to make it future proof in case another case is added below these.

Reply:
Yes, my original intention was as you said, let it go through the fall-through `default`. Adding the explicit `else` clause back (https://github.com/eclipse/omr/compare/830a971ddc30b106543f89e094b8e15fdd99e246..9f9f9e51f95a8382240a63edf9aad6e01d9f32d3)